shared/gcs/gcs.go

```diff
@@ -41,13 +41,25 @@ func Authenticate(ctx context.Context, serviceAccount string) error {
 	return err
 }
 
-// Exist checks if path exist under gcs bucket
-func Exist(ctx context.Context, bucketName, filePath string) bool {
-	handle := createStorageObject(bucketName, filePath)
+// Exist checks if path exist under gcs bucket,
+// this path can either be a directory or a file.
+func Exist(ctx context.Context, bucketName, storagePath string) bool {
+	// Check if this is a directory,
+	// gcs directory paths are virtual paths, they automatically got deleted if there is no child file
+	if len(ListChildrenFiles(ctx, bucketName, storagePath)) > 0 {
```

Comment:
My only concern is that `Exist` will be a costly operation, when usually it's perceived as cheap. Comments?

Reply:
Good point, and you'are absolutely right about this. I will explore options to make this operation cheaper

Reply:
updated, please review again